### PR TITLE
[WIP] Create a new build framework for Octokit using Cake.Frosting

### PR DESCRIPTION
**Pull request by ryangribble**
I've started work on #1440 to investigate moving to [CAKE](https://cakebuild.net/) for our build system.

I've decided to go down the road of using their latest [`Cake.Frosting`](https://github.com/cake-build/frosting) offering, which is essentially a .NET Core host/loader around the cake build framework.  This means we have full intellisense support of the build scripts AND the ability to debug the build scripts natively in Visual Studio!

So far it is building the `Octokit.Next` and `Octokit.Next.Tests` projects and running the unit tests.  I'm also experimenting with using GitVersion to calculate version numbers with the ultimate goal of publishing alpha packages to nuget on each merged PR to master as well as still supporting the official releases off a release branch

Everything lives under a `/build` sub directory.  To execute the build we simply `dotnet restore` and `dotnet run` in this directory.

The interesting parts are `BuildContext` which is essentially our build settings, `BuildLifetime` where we populate our settings from commandline options or other builtin helpers.  Then in the `/build/Tasks/` directory is where the tasks are actually implemented

In order to run via AppVeyor/Travis we will need `.ps1/.sh` bootstrap scripts, similar to the approach of the existing FAKE implementation

So far I have the following tasks planned:
- [x] Add Cake.Frosting build for Octokit.Next projects
- [ ] Add README.md file to `/build` directory to provide some info about the build script
- [x] Get running in AppVeyor
- [ ] ~~Get running TravisCI~~
- [ ] Get sourcelink replacement (GitLinkRunner) working
- [ ] Play around with GitVersion settings some more
- [x] Play around with nuget publishing
- [x] Decide whether to tackle the legacy octokit projects or only have this for dotnet core projects
- [ ] Get feedback from maintainers and contributors regarding the use of Cake and specifically Cake.Frosting

### CODE REVIEW

**Comment by M-Zuber** [comment]
While this would take away one of the main pushes I have to learn F#,
I think this will lower the bar for new contributors and therefore very welcome.

**Comment by shiftkey** [comment]
Assigning myself but I hope others can also :eyes: over the changes and get a feel for it all :metal:

File: build/Tasks/Build.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        // Don't run this task on OSX.
+        return context.Environment.Platform.Family != PlatformFamily.OSX;
+    }
+
+    public override void Run(BuildContext context)
+    {
+        foreach(var project in context.Projects)
+        {
+            context.Information("Building {0}...", project.Name);
+            context.DotNetCoreBuild(project.Path.FullPath, new DotNetCoreBuildSettings(){
```
Comment (M-Zuber) [inline build/Tasks/Build.cs]: In some places you kept the ()  when using object initialization syntax and others not

**Comment by ghuntley** [comment]
Yo @ryangribble check out https://github.com/octokit/octokit.net/issues/1440#issuecomment-261141143. If you want to bluejeans / chat about this stuff let me know.

**Comment by ryangribble** [comment]
Closing this off as it's now been incorporated into #1503

**Pull request closed without merging**